src/warp10-datasource.ts

```diff
@@ -183,24 +190,46 @@ export default class Warp10Datasource {
           throw new Error('Warp 10 expects the response to be a stack (an array), it isn\'t')
         }
 
-        // only one object on the stack, good user
-        if (res.data.length === 1 && typeof res.data[0] === 'object') {
-          let entries = []
-          res.data[0].forEach(key => {
-            entries.push({
-              text: key,
-              value: res.data[0][key]
-            })
-          })
-          return entries
+        // Grafana can handle different text/value for the variable drop list. User has three possibilites in the WarpScript result:
+        // 1 - let a list on the stack : text = value for each entry. 
+        // 2 - let a map on the stack : text = map key, value = map value. value will be used in the WarpScript variable.
+        // 3 - let some strings or numbers on the stack : it will be considered as a list, refer to case 1.
+        // Values could be strings or number, ignore other objects.
+
+        let entries = [];
+        if (1 == res.data.length && Array.isArray(res.data[0])) {
+          // case 1
+          res.data[0].forEach(elt => {
+            if (typeof elt === 'string' || elt instanceof String || typeof elt === 'number') {
+              entries.push({
+                text: elt.toString(),
+                value: elt.toString() // Grafana will turn every value to strings anyway !
+              })
+            }
+          });
+        } else if (res.data.length === 1 && typeof res.data[0] === 'object') {
+          // case 2
+          Object.keys(res.data[0]).forEach(key => {
```

Comment:
When you want both key and value of an object, you can use https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Object/entries

Reply:
That's what I did first entries().foreach( (k,v) =>{}), but it fails with the latest firefox. I don't want to add a polyfill for just this...